### DISCUSSION TITLE
getStaticPaths with only specific slugs

### DISCUSSION THREAD

**Discussion by gfmota** [vercel/next.js#53269] [Help]
### Summary

I'm working in a page that have the following route:
`/[step]/[another_slug]/[[...splat]]`
The step is a whole number from 0 to 6, and each step renders a different page, the another_slug has a minimal effect in the the render.
Previously I was using only static generation, what rendered the step 0 (default) in the build, this way, when someone enters any other step, the screen blinks the 0 step page.
I want to use getStaticPaths to define 7 different pages that got to be generated during the build, like this:
`
export const getStaticPaths = async () => {
    const paths = STEP_ARRAY.map((step) => ({ params: { step }, }));
    return { paths, fallback: 'blocking' }; 
};
`
But it breaks accusing that the other slugs can't be undefined.
Is there anyway to make it disconsider the other slug params and render the page in build time based only in the step slug?

**Comment by icyJoseph** [comment]
What if you return an empty array from getStaticPaths, paths, and set fallback to true.

Then when users navigate to the page, you validate the step to be a number in the range you want, and proceed as usual. It'll be a bit slow the first time, marginally, and unnoticeable if the page is prefetched via next/link.

You'd just need to handle the `useRouter().isFallback` case in the Page component.

And if the user is out of valid step number range, you return `{redirect: /0/foo/bar}` from GSP

Here for example, I render 3 pokemon, but keep the door open to render more:

- https://github.com/icyJoseph/next-js-react-gbg/blob/main/demo-app/src/pages/pokemon/%5Bid%5D.tsx

In your case you'd render zero, and let it all be generated as the users interact with your page.
